drivers/pwm/pwm_numaker.c

```diff
@@ -0,0 +1,574 @@
+		} else {
+			data->capture[pair].curr_edge_mode = EPWM_CAPTURE_INT_RISING_LATCH;
+			data->capture[pair].next_edge_mode = EPWM_CAPTURE_INT_RISING_LATCH;
+		}
+	} else {
+		data->capture[pair].pulse_capture = true;
+
+		if (flags & PWM_POLARITY_INVERTED) {
+			data->capture[pair].curr_edge_mode = EPWM_CAPTURE_INT_FALLING_LATCH;
+			data->capture[pair].next_edge_mode = EPWM_CAPTURE_INT_RISING_LATCH;
+		} else {
+			data->capture[pair].curr_edge_mode = EPWM_CAPTURE_INT_RISING_LATCH;
+			data->capture[pair].next_edge_mode = EPWM_CAPTURE_INT_FALLING_LATCH;
+		}
+	}
+
+	return 0;
+}
+
+static int pwm_numaker_enable_capture(const struct device *dev, uint32_t channel)
+{
+	const struct pwm_numaker_config *cfg = dev->config;
+	struct pwm_numaker_data *data = dev->data;
+	EPWM_T *epwm = cfg->epwm;
+	uint32_t pair = channel;
+	uint32_t channel_mask = (1 << channel);
+	uint32_t unit_time_nsec = (1000000000U / data->cycles_per_sec);
+
+	LOG_DBG("=== Enter %s ...\n", __func__);
+
+	if (!data->capture[pair].callback) {
+		LOG_ERR("PWM capture not configured");
+		return -EINVAL;
+	}
+
+	if (data->capture[pair].is_busy == true) {
+		LOG_ERR("Capture already active on this channel %d", pair);
+		return -EBUSY;
+	}
+
+	data->capture[pair].is_busy = true;
+
+	/* Set capture configuration */
+	EPWM_ConfigCaptureChannel(epwm, channel, unit_time_nsec, 0);
+
+	/* Enable Capture Function for EPWM */
+	EPWM_EnableCapture(epwm, channel_mask);
+
+	/* Enable Timer for EPWM */
+	EPWM_Start(epwm, channel_mask);
+
+	EPWM_ClearCaptureIntFlag(epwm, channel,
+				 EPWM_CAPTURE_INT_FALLING_LATCH | EPWM_CAPTURE_INT_RISING_LATCH);
+
+	/* EnableInterrupt */
+	EPWM_EnableCaptureInt(epwm, channel, data->capture[pair].curr_edge_mode);
+
+	LOG_DBG("===[%s] Channel=0x%x, CAPIEN=0x%x, CAPIF=0x%x", __func__, channel, epwm->CAPIEN,
+		epwm->CAPIF);
+
+	return 0;
+}
+
+static int pwm_numaker_disable_capture(const struct device *dev, uint32_t channel)
+{
+	const struct pwm_numaker_config *cfg = dev->config;
+	struct pwm_numaker_data *data = dev->data;
+	EPWM_T *epwm = cfg->epwm;
+	uint32_t channel_mask = (1 << channel);
+
+	LOG_DBG("=== Enter %s ...\n", __func__);
+
+	data->capture[channel].is_busy = false;
+	EPWM_Stop(epwm, channel_mask);
+	EPWM_ForceStop(epwm, channel_mask);
+	EPWM_DisableCapture(epwm, channel_mask);
+	EPWM_DisableCaptureInt(epwm, channel,
+			       EPWM_CAPTURE_INT_RISING_LATCH | EPWM_CAPTURE_INT_FALLING_LATCH);
+	EPWM_ClearCaptureIntFlag(epwm, channel,
+				 EPWM_CAPTURE_INT_FALLING_LATCH | EPWM_CAPTURE_INT_RISING_LATCH);
+	LOG_DBG("=== Enter %s ...CAPIEN=0x%x\n", __func__, epwm->CAPIEN);
+	return 0;
+}
+
+/*-----------------------------------------------------------------------------------
+ * Get capture cycles between current channel edge until next chnannel edge.
+ * The capture period counter down count from 0x10000, and auto-reload to 0x10000
+ *-----------------------------------------------------------------------------------
```

Comment:
Drop the `----------------` stuff, adds nothing to the comment, no need to get fancy, just write the comment. :-)